hera_pspec/pspecdata.py

```diff
@@ -1640,79 +1642,282 @@ def Jy_to_mK(self, beam=None):
+    err_msg = "dsets must be fed as a list of dataset string paths or UVData objects."
+    assert np.all([isinstance(d, UVData) for d in dsets]), err_msg
+
+    # configure polarization
+    if pol_pairs is None:
+        unique_pols = reduce(operator.and_, [set(d.polarization_array) for d in dsets])
+        pol_pairs = [(up, up) for up in unique_pols]
+
+    assert len(pol_pairs) > 0, "no pol_pairs specified"
+
+    # load beam
+    if isinstance(beam, (str, np.str)):
+        beam = pspecbeam.PSpecBeamUV(beam, cosmo=cosmo)
+
+    # beam and cosmology check
+    if beam is not None:
+        assert isinstance(beam, pspecbeam.PSpecBeamBase)
+        if cosmo is not None:
+            beam.cosmo = cosmo
+
+    # package into PSpecData
+    ds = PSpecData(dsets=dsets, wgts=[None for d in dsets], beam=beam)
+
+    # Rephase if desired
+    if rephase_to_dset is not None:
+        ds.rephase_to_dset(rephase_to_dset)
+
+    # perform Jy to mK conversion if desired
+    if Jy2mK:
+        ds.Jy_to_mK()
+
+    # Construct dataset pairs to operate on
+    if dset_pairs is None:
+        dset_pairs = list(itertools.combinations(range(Ndsets), 2))
+    if dset_labels is None:
+        dset_labels = ["dset{}".format(i) for i in range(Ndsets)]
+    err_msg = "dset_pairs must be fed as a list of len-2 integer tuples"
+    assert isinstance(dset_pairs, list), err_msg
+    assert np.all([isinstance(d, tuple) for d in dset_pairs]), err_msg
+
+    # Get baseline-pairs to use for each dataset pair
+    bls1_list, bls2_list = [], []
+    for i, dsetp in enumerate(dset_pairs):
+        # get bls if blpairs not fed
+        if blpairs is None:
+            (bls1, bls2, blps, xants1, 
+             xants2) = utils.calc_reds(dsets[dsetp[0]], dsets[dsetp[1]],
+                                       filter_blpairs=True,
+                                       exclude_auto_bls=exclude_auto_bls,
+                                       exclude_permutations=exclude_permutations,
+                                       group=group, Nblps_per_group=Nblps_per_group,
+                                       bl_len_range=bl_len_range)
+            bls1_list.append(bls1)
+            bls2_list.append(bls2)
+
+        # ensure fed blpairs exist in each of the datasets
+        else:
+            dset1_bls = dsets[dsetp[0]].get_antpairs()
+            dset2_bls = dsets[dsetp[1]].get_antpairs()
+            _bls1 = []
+            _bls2 = []
+            for _bl1, _bl2 in zip(bls1, bls2):
+                if (_bl1 in dset1_bls or _bl1[::-1] in dset1_bls) \
+                    and (_bl2 in dset2_bls or _bls2[::-1] in dset2_bls):
+                    _bls1.append(_bl1)
+                    _bls2.append(_bl2)
+
+            bls1_list.append(_bls1)
+            bls2_list.append(_bls2)
+
+    # Open PSpecContainer to store all output in
+    psc = container.PSpecContainer(filename, mode='rw')
+
+    # assign group name
+    if groupname is None:
```

Comment:
Not sure about this, but should we allow a prefix for groupname to be specified? This would let the user save power spectra into a given top-level group in the PSpecContainer, while maintaining the nice automatic `dset1_dset2` naming, e.g. to store spectra in `toplevel/dset1_dset2/...`.

Reply:
I think that's a good idea. in a different PR I'm going to actually slightly reformat the way I handle groupnames for the pspec pipeline, so I'll wait until then to make this change, but I'll keep it in mind while I work on it.